### DISCUSSION TITLE
New responsive UI model checking using functional programming

### DISCUSSION THREAD

**Discussion by Unsongheroes** [INTO-CPS-Association/into-cps-application#156] [Ideas]
Currently, we check the mm-model and configuration of the model when opening up the configuration, this is very much in line with the tutorials and is intended to stay that way. However, we do also use the same metric when we are check if the run simulation button should be able to run. This in itself is fine, but it causes a lot of errors that are difficult for the user to debug and understand due to faulty configuration or having the correct files but not the correct set-up. Errors with empty models can be "run". By implementing a more functional approach on how we verify the models at run-time instead of having a state to verify(where the state we verify against can be faulty) we can ensure proper information to the user, at better times and restrict access, and thereby faulty error messaging. 

We propose to implement run-time checks on the models and configuration during different parts of the user flow, and as a basis restrict users from areas of the system with understandable information as to why. alternatively, a live-updating "model-strength" indicator could be implemented to visually show the user if anything is not working and thereby not allowed for comprehension reasons.

**Comment by CThuleHansen** [comment]
go elm :P
